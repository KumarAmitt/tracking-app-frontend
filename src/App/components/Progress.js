import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';

import AppBar from './AppBar';
import { getProgressReport, getTotalPremium, loadDeals } from '../../store/slicers/deal';
import { getSessionInfo, loadSession } from '../../store/slicers/user_session';
import ProgressCircle from './ProgressCircle';
import { TARGET } from '../../constants';

const Progress = () => {
  const title = 'Progress Report';
  const dispatch = useDispatch();
  const sessionInfo = useSelector(getSessionInfo);
  const totalPremium = useSelector(getTotalPremium);
  const progressReport = useSelector(getProgressReport);
  const progressPercent = (totalPremium / TARGET) * 100;

  useEffect(() => {
    dispatch(loadDeals());
    dispatch(loadSession());
  }, []);

  // console.log('Total: ', totalPremium);
  // console.log('Progress: ', progressReport);

  // Object.entries(progressReport).forEach((p) => {
  //   console.log(p[0], p[1].map((e) => e.premium).reduce((a, b) => a + b));
  // });

  if (!sessionInfo.logged_in) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <AppBar title={title} />
      <div className="progress-stats">
        <div>
          <ProgressCircle value={progressPercent} />
          <div>Achieved</div>
        </div>
        <div>
          <ProgressCircle value={100 - progressPercent} />
          <div>Lag</div>
        </div>
      </div>
      <hr />
      <div>
        {
          Object.entries(progressReport).map((p) => (
            <div key={p[0]}>
              <div>{p[0]}</div>
              <div>{p[1].map((e) => e.premium).reduce((a, b) => a + b)}</div>
            </div>
          ))
        }
      </div>
    </>
  );
};

export default Progress;
