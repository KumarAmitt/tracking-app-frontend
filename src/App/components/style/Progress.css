.progress-stats {
  display: flex;
  justify-content: space-around;
  padding: 2em 0;
  margin-top: 3em;
}

.progress-stats > div {
  text-align: center;
}

.progress-item * + * {
  margin-top: 0.4em;
}

.progress-stats > * > *:first-child {
  transform: scale(1.3);
}

.progress-stats-label {
  font-size: 0.7rem;
  text-transform: uppercase;
  font-weight: bold;
  letter-spacing: 2px;
  margin-top: 0.8em;
}

.progress-items {
  background-color: #eee;
  padding: 1em;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 0.5em;
  height: 62vh;
  overflow: auto;
}

.progress-item {
  background-color: #fff;
  padding: 0.4em;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.progress-item .ins-type {
  font-size: 0.8rem;
  color: var(--accent-primary);
}

.progress-item .ins-premium {
  color: var(--action-color);
  font-size: 1.2rem;
  font-weight: bold;
  padding: 0.2em;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}

.progress-item .ins-count {
  font-size: 0.8rem;
  color: var(--accent-secondary);
}
