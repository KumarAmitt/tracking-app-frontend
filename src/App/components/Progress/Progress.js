import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import AppBar from '../AppBar/AppBar';
import { getProgressReport, getTotalPremium, loadDeals } from '../../../store/slicers/deal';

import ProgressCircle from '../utility/ProgressCircle';
import { TARGET } from '../../../constants';
import './style/Progress.css';
import { getUserInfo, loadSession } from '../../../store/slicers/user';

const Progress = () => {
  const title = 'Progress Report';
  const dispatch = useDispatch();
  const userInfo = useSelector(getUserInfo);
  const totalPremium = useSelector(getTotalPremium);
  const progressReport = useSelector(getProgressReport);
  const progressPercent = (totalPremium / TARGET) * 100;

  useEffect(() => {
    dispatch(loadSession());
    dispatch(loadDeals());
  }, []);

  if (userInfo.logged_in === false) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <AppBar title={title} />
      <div className="progress-stats">
        <div>
          <ProgressCircle value={progressPercent} color="primary" />
          <div className="progress-stats-label">Achieved</div>
        </div>
        <div>
          <ProgressCircle value={100 - progressPercent} color={progressPercent >= 100 ? 'primary' : 'secondary'} />
          <div className="progress-stats-label">Lag</div>
        </div>
      </div>
      <div className="progress-items">
        {
          Object.entries(progressReport).map((p) => (
            <div key={p[0]} className="progress-item">
              <div className="ins-type">{p[0]}</div>
              <div className="ins-premium">
                &#8377;
                {' '}
                {p[1].map((e) => e.premium).reduce((a, b) => a + b)}
              </div>
              <div className="ins-count">
                Quantity:
                {' '}
                <span>{p[1].length}</span>
              </div>
            </div>
          ))
        }
      </div>
    </>
  );
};

export default Progress;
