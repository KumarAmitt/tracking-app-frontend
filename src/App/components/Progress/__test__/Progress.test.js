import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import Progress from '../Progress';

const store = configureAppStore();

describe('Progress', () => {
  it('should match the snapshot of Progress component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Progress />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
