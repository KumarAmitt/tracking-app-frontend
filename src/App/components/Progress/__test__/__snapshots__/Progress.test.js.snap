// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should match the snapshot of Progress component 1`] = `
Array [
  <div
    className="appbar"
  >
    <a
      href="/add_deal"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
        />
      </svg>
    </a>
    <div
      className="title"
    >
      Progress Report
    </div>
  </div>,
  <div
    className="progress-stats"
  >
    <div>
      <div
        className="MuiBox-root MuiBox-root-1"
      >
        <div
          aria-valuenow={0}
          className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "transform": "rotate(-90deg)",
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={
                Object {
                  "strokeDasharray": "126.920",
                  "strokeDashoffset": "126.920px",
                }
              }
            />
          </svg>
        </div>
        <div
          className="MuiBox-root MuiBox-root-2"
        >
          <div
            className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
          >
            0%
          </div>
        </div>
      </div>
      <div
        className="progress-stats-label"
      >
        Achieved
      </div>
    </div>
    <div>
      <div
        className="MuiBox-root MuiBox-root-3"
      >
        <div
          aria-valuenow={100}
          className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-determinate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "transform": "rotate(-90deg)",
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={
                Object {
                  "strokeDasharray": "126.920",
                  "strokeDashoffset": "0.000px",
                }
              }
            />
          </svg>
        </div>
        <div
          className="MuiBox-root MuiBox-root-4"
        >
          <div
            className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
          >
            100%
          </div>
        </div>
      </div>
      <div
        className="progress-stats-label"
      >
        Lag
      </div>
    </div>
  </div>,
  <div
    className="progress-items"
  />,
]
`;
